### PR TITLE
implement new ServerData packet

### LINKED ISSUE #3363: MOTD reset to what defined in subserver when subserver close (IMPORTENT BUG!)
### Bungeecord version

1656

### Server version

1.19.1

### Client version

1.19.1

### Bungeecord plugins

nothing

### The bug

For some reason, when the sub server closes but the Bungeecord is still on, the bungeecord takes the MOTD of the sub server from the file server.properties and set him to the player when he kicked, but after a refresh, the MOTD went back to what was set in bungeecord.
video that explain the bug: https://www.youtube.com/watch?v=63HkJSNtfrI&ab_channel=L.P.C2

### Log output (links)

_No response_

### Checking

- [X] I am using BungeeCord and **not a fork**. Issues with forks should not be reported here.
- [X] I think this is **not** an issue with a bungeecord plugin.
- [ ] I have not read these checkboxes and therefore I just ticked them all.
- [X] This is not a question or plugin creation help request.

### PR DESCRIPTION
**Pull request by Outfluencer**
I implimented the new ServerData packet to drop it this fixes the behaviour of false motd's
Closes #3363

### CODE REVIEW

File: protocol/src/main/java/net/md_5/bungee/protocol/packet/ServerData.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+@EqualsAndHashCode(callSuper = false)
+public class ServerData extends DefinedPacket
+{
+
+    private String motd;
+    private String icon;
+    private boolean preview;
+    private boolean enforceSecure;
+
+
```
Comment (NoJokeFNA) [inline protocol/src/main/java/net/md_5/bungee/protocol/packet/ServerData.java:24]:
```suggestion
```

File: protocol/src/main/java/net/md_5/bungee/protocol/packet/ServerData.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+            buf.writeBoolean( false );
+        }
+
+        buf.writeBoolean( preview );
+
+        if ( protocolVersion >= ProtocolConstants.MINECRAFT_1_19_1 )
+        {
+            buf.writeBoolean( enforceSecure );
+        }
+
```
Comment (NoJokeFNA) [inline protocol/src/main/java/net/md_5/bungee/protocol/packet/ServerData.java:72]:
```suggestion
```

**Comment by md-5** [comment]
Some of these settings are per-server though. The correct behaviour would be to just update the motd

**Comment by Outfluencer** [comment]
now it should be the correct behaviour

**Comment by Brokkonaut** [comment]
I think it would be the best not to cancel the packet, but just set icon and motd to null if ping passthrough is disabled. When they are set to null the client does not update those entries in the server list and keeps the previous values.

**Comment by Outfluencer** [comment]
Isn't it better to just set it to null every time cause passthrough works with Forced Host and the server that is sending the packet 
can be any host

**Comment by Outfluencer** [comment]
works now..

**Pull request closed without merging**